DNN Platform/Library/Data/DataProvider.cs

```diff
@@ -3353,6 +3353,20 @@ public virtual DateTime GetDatabaseTime()
             return ExecuteScalar<DateTime>("GetDatabaseTime");
         }
 
+        public virtual DateTimeOffset GetDatabaseTimeOffset()
+        {
+            using (var reader = (SqlDataReader)ExecuteSQL("SELECT SYSDATETIMEOFFSET()"))
+            {
+                if(reader.HasRows && reader.Read())
+                {
+                    return reader.GetDateTimeOffset(0);
+                } else
+                {
+                    throw new Exception("Error retrieving database UTC offset");
```

Comment:
I don't think this will ever happen and even if it does, let's keep the original exception info and stack trace better